app/lib/metrics_service.rb

```diff
@@ -0,0 +1,108 @@
+# Service to record events to Mixpanel via Bard
+#
+# Mixpanel is a business analytics service for tracking user interactions.
+# Bard is a DSP service that mediates writes to Mixpanel.
+#
+# Most logging to Mixpanel is done in JavaScript; see metrics-api.js and
+# scp-api-metrics.js in /app/javascript/lib/.  But some logging is done
+# in Rails to make it easy to analyze metrics around long-running events
+# triggered by user interactions -- like upload and ingest -- or special
+# identity resolution.
+
+class MetricsService
+
+  BARD_ROOT = Rails.application.config.bard_host_url
+
+  def self.post_to_bard(params, user)
+    params.merge!({
+      method: 'POST'
+    })
+    begin
+      Rails.logger.info "#{Time.zone.now}: Posting to Mixpanel.  Params: #{params}"
+      # Uncomment line below to get known-good test data
+      # puts "Posting to Mixpanel.  Params: #{params}"
+      RestClient::Request.execute(params)
+    rescue RestClient::ExceptionWithResponse => e
```

Comment:
We might actually want to use `RestClient::Exception` here, as there could be a scenario where Bard responds with simply an error code (503, 400) and doesn't have a response body.  In that case, I'm don't think that it would be caught here.

Reply:
Done in 6265f13.